dropwizard-lifecycle/src/test/java/io/dropwizard/lifecycle/setup/ExecutorServiceBuilderTest.java

```diff
@@ -124,6 +127,41 @@ public void shouldUseInstrumentedThreadFactory() {
         assertThat(castedExec.getThreadFactory()).isInstanceOf(InstrumentedThreadFactory.class);
     }
 
+    @Test
+    public void nameWithoutFormat() {
+        final String[][] tests = new String[][] {
+            { "my-client-%d", "my-client" },
+            { "my-client--%d", "my-client-" },
+            { "my-client-%d-abc", "my-client-abc" },
+            { "my-client%d", "my-client" },
+            { "my-client%d-abc", "my-client-abc" },
+            { "my-client%s", "my-client" },
+            { "my-client%sabc", "my-clientabc" },
+            { "my-client%10d", "my-client" },
+            { "my-client%10d0", "my-client0" },
+            { "my-client%-10d", "my-client" },
+            { "my-client%-10d0", "my-client0" },
+            { "my-client-%10d", "my-client" },
+            { "my-client-%10dabc", "my-clientabc" },
+            { "my-client-%1$d", "my-client" },
+            { "my-client-%1$d-abc", "my-client-abc" },
+            { "-%d", "" },
+            { "%d", "" },
+            { "%d-abc", "-abc" },
+            { "%10d", "" },
+            { "%10dabc", "abc" },
+            { "%-10d", "" },
+            { "%-10dabc", "abc" },
+            { "%10s", "" },
+            { "%10sabc", "abc" },
```

Comment:
Try a bunch of different combinations of format strings. Let me know if I've missed combinations that would be useful to test.